### ISSUE TITLE
How to set Proxy server

### ISSUE THREAD

**Issue by MasonAlice** [fork-dev/TrackerWin#1953]
I just want to config a proxy server for fork, instead of setting global proxy config for git.

**Comment by DanPristupov** [comment]
Why is that important? What is the use case?

**Comment by HowellDong** [comment]
Because I am in China, github can only be accessed through proxies!

**Comment by MasonAlice** [comment]
> Why is that important? What is the use case?

Because I am in China, github can only be accessed through proxies!

**Comment by DanPristupov** [comment]
Then configure proxy in the command line: https://github.com/fork-dev/TrackerWin/issues/220

**Comment by gobater** [comment]
@DanPristupov , how does gitfork access to gravatar.com?
The gravatar's stopped working in fork. Not sure if it might be due to the corporate proxy? (access to git is fine)

**Comment by DanPristupov** [comment]
@gobater 

> how does gitfork access to gravatar.com?

I just checked the source code:

```
var client = new System.Net.WebClient();
client.DownloadDataCompleted += (sender, args) => { /* */}
client.DownloadDataAsync(GravatarUri(email));
```

> The gravatar's stopped working in fork. Not sure if it might be due to the corporate proxy? (access to git is fine)

It could be. Looks like WebClient doesn't use system proxy by default.

https://stackoverflow.com/a/21183464

As a test, try to add to `%localappdata%\fork\app-1.92.0\Fork.exe.config`:

```
<system.net>
    <defaultProxy useDefaultCredentials="true" />
</system.net>
```

**Comment by gobater** [comment]
Hi @DanPristupov ,

Now it works. It's curious that if I remove the Proxy setting in the config and restart fork, I still see the avatars. Are they cached?

**Comment by DanPristupov** [comment]
> It's curious that if I remove the Proxy setting in the config and restart fork, I still see the avatars. Are they cached?

No, not intentionally, at least. May be there is a cache on the system level, but Fork does not cache avatars between sessions.

**Comment by gobater** [comment]
Just had some time to further detail the issue...

My enterprise uses a proxy, which is setup via PAC. When I'm not connected to the enterprise network, PAC configuration fails, and many tools fallback to "no-proxy" (e.g. chrome, edge, firefox... which all are configured to use the system settings). However, in this case, Git Fork fails to connect to gravatar (independently of the settings in Fork.exe.config)

As soon as I connect to the enterprise network (e.g. via VPN) Git Fork connects to gravatar without issues (independently of the settings in Fork.exe.config) --> so the real problem I'm facing is falling back to "no-proxy" when the configured proxy cannot be reached or when PAC cannot be reached...

**Comment by brandonh-msft** [comment]
I would like to see a way to do this outside of the netfx app config file. Requiring this method means every time Fork updates, I have to go into the new `app-<version>` folder and make the same edit to it.

Additionally, for proxies that require certs this doesn't appear to work. I have the CA Bundle specified in my global git config, but Fork cannot pull, push, fetch from the remote repositories. Git in terminal, however, can just fine.

For reference, here are the `http` settings I have set:

~~~
http.sslbackend=openssl
http.proxy=http://<my-corp-proxy>:8080
http.sslcainfo=C:/Users/<me>/cacert.mycorp.pem
https.proxy=http://<my-corp-proxy>:8080
~~~

In Fork on a `fetch` I get

~~~
fatal: unable to access 'https://github.com/my-corp/my-repo.git/': schannel: CertGetCertificateChain trust error CERT_TRUST_REVOCATION_STATUS_UNKNOWN
~~~

while `git fetch` works just fine from terminal

**Comment by alexandrejobin** [comment]
Any news on this? I have to add `<defaultProxy useDefaultCredentials="true" />` to the config file but as soon as the app is updated, the config file is reseted to I need to add it back again.

Any way to make it permanent or use the default proxy in windows?

**Comment by DanPristupov** [comment]
We added the `useDefaultCredentials` proxy setting in Fork 2.14.

**Comment by alexandrejobin** [comment]
Works great with 2.14! Thank you!
